src/Common/Thermodynamics/relations.jl

```diff
@@ -1967,3 +1969,80 @@ relative_humidity(ts::ThermodynamicState{FT}) where {FT <: Real} =
         typeof(ts),
         PhasePartition(ts),
     )
+
+"""
+    total_specific_enthalpy(e_tot, R_m, T)
+
+Total specific enthalpy, given
+ - `e_tot` total specific energy
+ - `R_m` [`gas_constant_air`](@ref)
+ - `T` air temperature
+"""
+function total_specific_enthalpy(
+    e_tot::FT,
+    R_m::FT,
+    T::FT
+) where {FT <: Real}
+    return e_tot + R_m * T
+end
+
+"""
+    total_specific_enthalpy(ts)
+
+Total specific enthalpy, given
+ - `e_tot` total specific energy
+ - `ts` a thermodynamic state
+"""
+function total_specific_enthalpy(
+    ts::ThermodynamicState{FT},
+    e_tot::FT
+) where {FT <: Real}
+    R_m = gas_constant_air(ts)
+    T = air_temperature(ts)
+    return total_specific_enthalpy(e_tot, R_m, T)
+end
+
+"""
+    specific_enthalpy(e_tot, R_m, T)
+
+Specific enthalpy, given
+ - `e_int` internal specific energy
+ - `R_m` [`gas_constant_air`](@ref)
+ - `T` air temperature
+"""
+function specific_enthalpy(
+    e_int::FT,
+    R_m::FT,
+    T::FT
+) where {FT <: Real}
+    return e_int + R_m * T
+end
+
+"""
+    specific_enthalpy(ts)
+
+Specific enthalpy, given a thermodynamic state `ts`.
+"""
+function specific_enthalpy(ts::ThermodynamicState{FT}) where {FT <: Real}
+    e_int = internal_energy(ts)
+    R_m = gas_constant_air(ts)
+    T = air_temperature(ts)
+    return specific_enthalpy(e_int, R_m, T)
+end
+
+"""
+    moist_static_energy(ts, e_kin, e_tot)
```

Comment:
Maybe this one is easier to compute as `specific_enthalpy + geopotential`, so you do not need deal with kinetic energy but can stay with thermodynamic and potential energies only?